### PR TITLE
[cancel_trajectory] Operation to cancel the current trajectory

### PR DESCRIPTION
**Pull request by Rauldg**

### CODE REVIEW

**Comment by Rauldg** [comment]
@skasperski  

After the cancel trajectory operation is called, the trajectory follower switches to the state trajectory_finished. Which is fine in my opinion.

File: tasks/Task.cpp [hunk 1]
```diff
@@ -194,3 +194,18 @@ void Task::cleanupHook()
...
+{
+    LOG_INFO_S << "Current trajectory cancellation requested";
+    // try to remove the trajectory and if no exception occurs, return true
+    try {
+        trajectoryFollower.removeTrajectory();
+    } catch (std::runtime_error& e) {
+        LOG_ERROR_S << "Could not remove trajectory: " << e.what();
+        return false;
+    }
+    trajectoryFollower.removeTrajectory();
```
Comment (skasperski) [inline tasks/Task.cpp]: Is it intentional that removeTrajectory() is called twice here?

File: tasks/Task.cpp [hunk 2]
```diff
@@ -194,3 +194,18 @@ void Task::cleanupHook()
...
+    trajectories = std::vector<SubTrajectory>();
```
Comment (skasperski) [inline tasks/Task.cpp]: Maybe just call "trajectories.clear()" here.

File: tasks/Task.cpp (hunk 1 above)
Comment (Rauldg) [reply tasks/Task.cpp]: No, That is indeed a mistake.

**Comment by Rauldg** [comment]
@skasperski I have updated the PR according to the feedback, thanks for the improving recommendations

**Pull request merged**